packages/store/src/symbols.ts

```diff
@@ -147,3 +153,13 @@ export interface NgxsAfterBootstrap {
 }
 
 export type NgxsLifeCycle = Partial<NgxsOnInit> & Partial<NgxsAfterBootstrap>;
+
+export type FunctionCall = <T>(func: (...args: any[]) => T) => T;
+
+/*
+ * Execution strategy interface
+ */
+export interface NgxsExecutionStrategy {
+  enter<T>(func: (...args: any[]) => T): T;
```

Comment:
why do you do not use everywhere your FunctionCall type ?

Reply:
I think it gave me problems. Will look again.

Reply:
Got rid of this because it is not used. Made signature simpler so that this is not necessary as well.